clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -988,12 +988,11 @@ public List<PartitionInfo> partitionsFor(String topic, Duration timeout) {
                 throw new TimeoutException();
             }
 
-            final TopicMetadataApplicationEvent topicMetadataApplicationEvent =
-                    new TopicMetadataApplicationEvent(topic, timeout.toMillis());
-            wakeupTrigger.setActiveTask(topicMetadataApplicationEvent.future());
+            final TopicMetadataEvent topicMetadataEvent = new TopicMetadataEvent(Optional.of(topic), timeout.toMillis());
+            wakeupTrigger.setActiveTask(topicMetadataEvent.future());
```

Comment:
A minor internal change was to use an `Optional` for the `TopicMetadataEvent` constructor to indicate "all topics" (`Optional.empty()`) vs a specific topic (`Optional.of(topic)`).

Reply:
Done.